src/operator/tensor/dot-inl.cuh

```diff
@@ -1053,7 +1053,7 @@ inline void DotDnsCsrDnsImpl(const OpContext& ctx, const gpu& gpu_dev,
   TBlob csr_indices = rhs.aux_data(csr::kIdx);
   TBlob csr_indptr = rhs.aux_data(csr::kIndPtr);
   if (!rhs.storage_initialized()) {
-    FillZerosCsrImpl(s, *ret);
```

Comment:
Why is there segmentation fault instead of log fatal err msg? The CheckAndAllocData function used by FillZerosCsrImpl does check stype: https://github.com/apache/incubator-mxnet/blob/master/include/mxnet/ndarray.h#L621

Reply:
Maybe it's failing in set_aux_shape() here: https://github.com/apache/incubator-mxnet/blob/b2ccd34ad2801b6c87c957c28ad718562a4c5b6e/src/operator/tensor/init_op.h#L359? I can do a check quickly and let you know the result.